### ISSUE TITLE
Bug: Exception - imagecopyresampled():

### ISSUE THREAD

**Issue by richarda253** [moodle-an-hochschulen/moodle-theme_boost_union#304]
Everything seems to be functioning OK but I am seeing a lot of the following entries in the Apache error logs for Moodle41. and 4.2 sites,

========================
Moodle 4.1.3+ (Build: 20230519)
Boost Union v4.1-r7
PHP 8.1.18

Default exception handler: 
Exception - imagecopyresampled(): 
Argument #2 ($src_image) must be of type GdImage, bool given 
Debug: 
Error code: generalexceptionmessage
 line 53 of /lib/gdlib.php: TypeError thrown
 line 53 of /lib/gdlib.php: call to imagecopyresampled()
 line 393 of /lib/gdlib.php: call to imagecopybicubic()
 line 1132 of /lib/filestorage/stored_file.php: call to resize_image_from_image()
 line 364 of /theme/boost_union/lib.php: call to stored_file->resize_image()
 line 5274 of /lib/filelib.php: call to theme_boost_union_pluginfile()
 line 44 of /pluginfile.php: call to file_pluginfile()
========================
and
========================
Moodle 4.2+ (Build: 20230519)
Boost Union v4.1-r7
PHP 8.1.18

Default exception handler: 
Exception - imagecopyresampled(): 
Argument #2 ($src_image) must be of type GdImage, bool given 
Debug: 
Error code: generalexceptionmessage
 line 53 of /lib/gdlib.php: TypeError thrown
 line 53 of /lib/gdlib.php: call to imagecopyresampled()
 line 393 of /lib/gdlib.php: call to imagecopybicubic()
 line 1146 of /lib/filestorage/stored_file.php: call to resize_image_from_image()
 line 364 of /theme/boost_union/lib.php: call to stored_file->resize_image()
 line 5286 of /lib/filelib.php: call to theme_boost_union_pluginfile()
 line 44 of /pluginfile.php: call to file_pluginfile()
========================

The sites' seem to be functioning OK.,  If I switch the theme to Boost, the error messages stop being generated.

**Comment by danowar2k** [comment]
Looking at the Moodle code, the function resize_image() in lib/filestorage/stored_file.php does this:

```
        // Create a new image from the file.
        $original = @imagecreatefromstring($content);
```

That's the spot where imagecreatefromstring returns a bool value which isn't understood later.
That functions returns a bool value (i.e. false), "if the image type is unsupported, the data is not in a recognised format, or the image is corrupt and cannot be loaded."

So, basically, if you find out which image theme_boost_union is failing to resize (going by the line number 364, it's one from the fileareas 'logo', 'logocompact' or 'favicon', we can continue the analysis. You could list the files in those fileareas, for a start.

**Comment by abias** [comment]
@richarda253 - I would assume that you used an .ico file as Favicon in Boost Union. Until recently, these files triggered this error. This was reported and fixed in #387.

I will close this issue now. Feel free to open it again if you become aware that a different image is causing this error for you.
